wagtail/snippets/tests/test_edit_view.py

```diff
@@ -730,6 +730,68 @@ def test_edit_snippet_with_revision(self):
         self.assertEqual(log_entries.count(), 1)
         self.assertEqual(log_entries.first().revision, revision)
 
+    def test_overwrite_revision_with_json_response(self):
+        self.test_snippet.text = "Initial revision"
+        revision = self.test_snippet.save_revision(user=self.user)
+        self.assertEqual(self.test_snippet.revisions.count(), 1)
+        response = self.post(
+            post_data={
+                "text": "Updated revision",
+                "overwrite_revision_id": revision.pk,
+            },
+            headers={"Accept": "application/json"},
+        )
+
+        # Should be a 200 OK JSON response
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response["Content-Type"], "application/json")
+        self.assertEqual(response.json(), {"success": True, "pk": self.test_snippet.pk})
+
+        self.assertEqual(self.test_snippet.revisions.count(), 1)
+        revision.refresh_from_db()
+        self.assertEqual(revision.content["text"], "Updated revision")
+
+    def test_overwrite_non_latest_revision(self):
```

Comment:
Can we have another test for a `overwrite_revision_id` that's nonexistent, for completeness sake?